# Types of NGS



|            | Advantages                                                   | Limitations                                                  | Examples                                                     |
| ---------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
| Short-Read | · Higher sequence fidelity<br/>· Cheap  <br/>· Can sequence fragmented DNA | · Not able to resolve structural variants, phasing alleles or distinguish highly homologous genomic regions<br />· Unable to provide coverage of some repetitive regions | * Ion Torrent<br />* 454<br />* Illumina<br />* SOLiD<br />* cPAS<br />* MPSS<br /> |
| Long-Read  | · Able to sequence genetic regions that are difficult to characterize with short-read seq due to repeat sequences<br />· Able to resolve structural rearrangements or homologous regions<br />· Able to read through an entire RNA transcript to determine the specific isoform<br />· Assists *de novo* genome assembly | · Lower per read accuracy<br />· Bioinformatic challenges, caused by coverage biases, high error rates in base allocation, scalability and limited availability of appropriate pipelines | Pac-Bio (2nd)<br />* Single molecule real time sequencing (3rd Gen)<br />Nanopore (3rd Gen) |

(adapted from [technologynetworks.com](https://www.technologynetworks.com/genomics/articles/an-overview-of-next-generation-sequencing-346532), [wikipedia.org](https://en.wikipedia.org/wiki/DNA_sequencing))



Also: 

* Whole-genome vs Whole-exome sequencing

* Whole-exome (DNA sequencing) vs RNA-Seq (RNA sequencing)

  

---

Prev: [06-Next-Generation-Sequencing.md](06-Next-Generation-Sequencing.md) 

Next:  [08-Appendix.md](08-Appendix.md) 

